server/src/main/java/org/elasticsearch/health/HealthService.java

```diff
@@ -45,10 +49,23 @@ private static HealthComponentResult createComponentFromIndicators(List<HealthIn
         assert indicators.size() > 0 : "Component should not be non empty";
         assert indicators.stream().map(HealthIndicatorResult::component).distinct().count() == 1L
             : "Should not mix indicators from different components";
+        Set<String> duplicateIndicatorNames = findDuplicatesByName(indicators);
```

Comment:
can we call this inside the assert to avoid test only code be executed in production?

Reply:
Done